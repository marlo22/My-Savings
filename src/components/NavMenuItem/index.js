import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

import { Text } from 'react-native';
import { ListItem, Icon } from 'native-base';

const MenuItem = styled(ListItem)`
  border-bottom-width: 0;
  margin: 0;
  padding: 15px;
  border-bottom-width: 2px;
  border-bottom-color: rgba(255, 255, 255, 0.4);
`;

const IconWrapper = styled.View`
  background-color: #ffffff;
  width: 30px;
  height: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  margin-right: 10px;
`;

const NavMenuItem = ({ onPress, iconName, text }) => (
  <MenuItem onPress={onPress}>
    <IconWrapper>
      <Icon name={iconName} />
    </IconWrapper>
    <Text>{text}</Text>
  </MenuItem>
);

NavMenuItem.propTypes = {
  onPress: PropTypes.func.isRequired,
  iconName: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default NavMenuItem;