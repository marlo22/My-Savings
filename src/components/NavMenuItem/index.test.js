import React from 'react';
import { shallow } from 'enzyme';

import NavMenuItem from './';

const onPress = jest.fn();
const iconName = 'smile';
const text = 'Hello world';

const component = shallow(
  <NavMenuItem
    onPress={onPress}
    iconName={iconName}
    text={text}
  />
);

describe('<NavMenuItem />', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should set passed text', () => {
    const textComponent = component.find('Text');
    expect(textComponent.prop('children')).toBe(text);
  });

  test('should set passed icon', () => {
    const icon = component.find('Styled(Icon)');
    expect(icon.prop('name')).toBe(iconName);
  });
});