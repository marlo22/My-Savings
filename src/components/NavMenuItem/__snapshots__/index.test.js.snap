// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMenuItem /> snapshot 1`] = `
<Styled(Styled(ListItem))
  onPress={[MockFunction]}
>
  <Styled(View)>
    <Styled(Icon)
      name="smile"
    />
  </Styled(View)>
  <Text>
    Hello world
  </Text>
</Styled(Styled(ListItem))>
`;
